rust/candid/src/de.rs

```diff
@@ -254,6 +255,19 @@ impl<'de> Deserializer<'de> {
         }
         Ok(())
     }
+    fn is_zero_sized_type(&self, t: &Type) -> bool {
+        match t.as_ref() {
+            TypeInner::Null | TypeInner::Reserved => true,
+            TypeInner::Record(fs) => {
+                fs.is_empty()
+                    || fs.iter().all(|f| {
+                        let t = self.table.trace_type(&f.ty).unwrap();
+                        self.is_zero_sized_type(&t)
+                    })
+            }
+            _ => false,
+        }
+    }
```

Comment:
Also, is this safe against recursive records, which could cause an infinite loop? Or have recursive records been ruled out before somewhere?

Reply:
Yes, it's safe. Recursive records have been replaced with empty when parsing the table type. I add a comment in the code.

Reply:
Great, thanks!